cypress/integration/search-workout-templates.spec.ts

```diff
@@ -8,10 +8,22 @@ describe('Search workout templates', () => {
      * 6. Go back to workout programs
      */
 
-    it('user can search for a workout template', () => {
+    it('can navigate to search page', () => {
         cy.visit('/');
         cy.findByRole('button', {
             name: /Search Workouts/i,
         }).click({ force: true });
     });
+
+    it('can search for a workout template', () => {
+        cy.get('input[data-cy="search-workout-input"]')
```

Comment:
be wary of using custom selectors to find elements. is cypress able to find by `role`? or find by `text`? or find by something a user would use to interact with the button?

Reply:
Interestingly enough, I think they recommend using selectors:
https://docs.cypress.io/guides/references/best-practices

Thoughts?

Reply:
So, kind of.  

In the old days, automated testing solutions used css class names, ids, or XPath. What they are getting at there by suggesting a `data-` attribute isn't wrong. It is useful to be able to grab elements via a `data` attribute. However, users don't use your app using data attributes. This is taking a page out of `testing-library`'s stance. https://testing-library.com/docs/guiding-principles.

TL;DR If you want to test your app the way users use it, you should test your app the way users use it. 😉

Reply:
That makes sense. I'll refactor that when I get a chance.